---
title: "1102-MoreSamples"
author: "Mac Campbell"
date: "August 19, 2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

```{r}
library(tidyverse)
library(viridis)
library(ggpubr)
library(RcppCNPy)
```

## Campbell et al. (2021) samples

Can my other samples be integrated here? Renaming, see 1102.1-rename.sh,

Aligning to /home/maccamp/mccloud-rrt/genome/omyV6Chr.fasta using run_align.sh from Ensieh.    


## Remove low-ballers at some thresh.
(base) maccamp@farm:~/data/trout-rad/BigCk$

ls | grep sort.flt.bam| grep -v mapped > bams
bash samtools_flagstat.sh bams 
cat *mapped_bams | grep "mapped (" | awk -F" " '{print $1}' > counts
 paste bams counts  > stats.txt

Scott-M021553_M209_6F_RA.sort.flt.bam	2149592 #Adult Anadromous
Scott-M021554_M209_6G_RA.sort.flt.bam	626813 #Adult Anadromous
Scott-M021604_M210_1A_RA.sort.flt.bam	35765 #Juvenile/resident
Scott-M021605_M210_1B_RA.sort.flt.bam	251125
Scott-M021606_M210_1C_RA.sort.flt.bam	1891084
Scott-M021607_M210_1D_RA.sort.flt.bam	741100
Scott-M021616_M210_2E_RA.sort.flt.bam	646582
Scott-M021617_M210_2F_RA.sort.flt.bam	914955
Scott-M021618_M210_2G_RA.sort.flt.bam	206444
Scott-M021619_M210_2H_RA.sort.flt.bam	419690
Scott-M021624_M210_3E_RA.sort.flt.bam	527048
Scott-M021625_M210_3F_RA.sort.flt.bam	295583
Scott-M021626_M210_3G_RA.sort.flt.bam	735745
Scott-M021627_M210_3H_RA.sort.flt.bam	1049920
Scott-M021628_M210_4A_RA.sort.flt.bam	874565
Scott-M021629_M210_4B_RA.sort.flt.bam	630891
Scott-M021630_M210_4C_RA.sort.flt.bam	914311
Scott-M021631_M210_4D_RA.sort.flt.bam	978366
Scott-M021636_M210_5A_RA.sort.flt.bam	515182
Scott-M021637_M210_5B_RA.sort.flt.bam	418973
Scott-M021638_M210_5C_RA.sort.flt.bam	791119
Scott-M021639_M210_5D_RA.sort.flt.bam	934781
Scott-M021644_M210_6A_RA.sort.flt.bam	784776
Scott-M021645_M210_6B_RA.sort.flt.bam	607840
Scott-M021646_M210_6C_RA.sort.flt.bam	735162
Scott-M021647_M210_6D_RA.sort.flt.bam	283009

Let's try 200K reads, gives us 313 total fish. We may be able to drop sampling to a few individuals (5?) from each location so that we match the general sampling from Habibi et al., so top five from this for a trial + the top steelhead.
Scott-M021606_M210_1C_RA.sort.flt.bam	1891084
Scott-M021627_M210_3H_RA.sort.flt.bam	1049920
Scott-M021631_M210_4D_RA.sort.flt.bam	978366
Scott-M021639_M210_5D_RA.sort.flt.bam	934781
Scott-M021617_M210_2F_RA.sort.flt.bam	914955
Scott-M021553_M209_6F_RA.sort.flt.bam	2149592 #Adult Anadromous

294.csv for one chrom     

```{sh, eval=FALSE}
srun -p high -t 48:00:00 --mem=32G --nodes=2 $HOME/angsd/angsd -P 24  -bam bamlists/294.bamlist -minInd 259 -GL 1 -ref genome/omyV6Chr.fasta \
-doGLF 2 -doMajorMinor 1 -doMaf 2 -SNP_pval 1e-6 -minMapQ 20 -minQ 20 -r omy01: \
-out outputs/1102/294 > outputs/1102/beagle.out 2> outputs/1102/beagle.err &

python $HOME/pcangsd/pcangsd.py -beagle outputs/1102/294.beagle.gz -admix -o outputs/1102/294 -threads 10
```

Moving in more samples here:    
```{r}
#meta<-read_csv("meta/294.csv")
#meta<-read_csv("meta/299.csv")
meta<-read_csv("meta/318.csv")
```

### Summarize Meta

```{r}
bars<-meta %>% group_by(Group, MajorGroup) %>% summarize(Count=n()) %>% select(MajorGroup, Group, Count) %>% arrange(MajorGroup)


```



```{r, echo=FALSE, warning=FALSE, message=FALSE}
# Some shameless copying from Eric A.
#' @param samples character vector with the individuals IDs in the order in which
#' they were passed in the bamlist to angsd.
#' @param cov covariance matrix
covar2pcs <- function(samples, cov) {
  
  
  eig <- eigen(cov, symm = TRUE)
  PC <- as.data.frame(eig$vectors) %>%
    as_tibble() %>%
    setNames(sprintf("PC-%02d", 1:ncol(.)))
  
  samtib <- tibble(sample = samples)
  
  list(
    PCs = bind_cols(samtib, PC),
    eigevalues = eig$values
  )
}
```

```{r, warning=FALSE, message=FALSE}

cov<-read_delim("outputs/1102/318.cov", col_names=FALSE, delim=" ") %>% as.matrix()

```

```{r}
pca <- covar2pcs(meta$Bamlist, cov)

pca_long <- pca$PCs %>%
  tidyr::gather(., key = "PC", "val", -sample)

# then expand a grid of the possible comparisons (ordered)
expg <- expand.grid(sample = pca$PCs$sample,
                    PCx = sprintf("PC-%02d", 1:6),
                    PCy = sprintf("PC-%02d", 1:6),
                    stringsAsFactors = FALSE) %>%
  tibble::as_tibble()

# then left join the pca results onto that
pca_pairs <- dplyr::left_join(expg, pca_long, by = c("sample", "PCx" = "PC")) %>%
  dplyr::rename(val_x = val) %>%
  dplyr::left_join(pca_long, by = c("sample", "PCy" = "PC")) %>%
  dplyr::rename(val_y = val)

pp_meta <- pca_pairs %>%   # just keep the first 6 PCs around
  left_join(., meta, by = c("sample" = "Bamlist"))

# now, that has the first 6 PCs in it.  If we want to focus on the just the
# first 3, we could do 
npc <- 3
pp_meta2 <- pp_meta %>%
  filter( (PCx %in% sprintf("PC-%02d", 1:npc)) & 
            (PCy %in% sprintf("PC-%02d", 1:npc)) )

ggplot(pp_meta2, aes(x = val_x, y = val_y, fill = Group)) +
  geom_point(pch = 21, size = 2) +
  scale_fill_discrete(na.value = "white") + 
  facet_grid(PCy ~ PCx, scales = "free")


```
```{r}

eig <- eigen(cov, symm = TRUE)
var<-eig$values/sum(eig$values)
cumvar<-cumsum(eig$values)/sum(eig$values)

head(var)
head(cumvar)
```


```{r}
sub12<-pp_meta2 %>% filter( (PCx =="PC-01") & (PCy =="PC-02") )

pc12<-ggplot(sub12, aes(x = val_x, y = val_y, fill = Group, shape=MajorGroup)) +
  geom_point(size = 2, alpha=0.75) +
  scale_fill_discrete(na.value = "white") + 
  theme_bw()+
  theme(panel.grid=element_blank())+
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = ""))+
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = ""))+
  scale_fill_viridis_d(option = "magma") +
  scale_shape_manual(values = c(21,22,22,23,24,25)) +
  theme(legend.position = "")+
  theme(plot.title = element_text(hjust = 0.5)) 

sub13<-pp_meta2 %>% filter( (PCx =="PC-01") & (PCy =="PC-03") )

pc13<-ggplot(sub13, aes(x = val_x, y = val_y, fill = Group, shape=MajorGroup)) +
  geom_point(size = 2, alpha=0.75) +
  scale_fill_discrete(na.value = "white") + 
  theme_bw()+
  theme(panel.grid=element_blank())+
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = ""))+
  ylab(paste("PC3", " ", round((100*var[3]),2), "%", sep = ""))+
 # theme(legend.position = "")+
  theme(plot.title = element_text(hjust = 0.5)) +
  scale_fill_viridis_d(option = "magma") +
  scale_shape_manual("Major Group", values = c(21,22,22,23,24,25)) +
  guides(fill = guide_legend(override.aes=list(shape=15, color= viridis(8, option="magma")))) 

pcs<-ggarrange(pc12, pc13, ncol = 2, widths=c(1.1, 1.5))
pcs

ggsave("outputs/1102/pcs.pdf", width=8, height=4.5)

```

```{r}
fig<-ggplot(sub12, aes(x = val_x, y = val_y, fill = Group, shape=MajorGroup)) +
  geom_point(size = 2, alpha=0.75) +
  scale_fill_discrete(na.value = "white") + 
  theme_bw()+
  theme(panel.grid=element_blank())+
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = ""))+
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = ""))+
  scale_fill_viridis_d(option = "magma") +
  scale_shape_manual(values = c(21,22,22,23,24,25)) +
  theme(plot.title = element_text(hjust = 0.5)) 

fig
```

## Pull out REDB and MCRT

```{r}
ggplot(pp_meta %>% filter(Group %in% c("REDB","MCRT")), aes(x = val_x, y = val_y, shape = Group, color=Watershed)) +
  geom_point(size = 2, alpha=0.75) +
  scale_fill_discrete(na.value = "white") + 
  facet_grid(PCy ~ PCx, scales = "free") 

ggsave("outputs/1102/REDB-pca.pdf", width=14, height=12)
```

## Pull out RAINB

```{r}
ggplot(pp_meta %>% filter(Group %in% c("HRNB","CRT","EGLK")), aes(x = val_x, y = val_y, shape = Group, color=Watershed)) +
  geom_point(size = 2, alpha=0.75) +
  scale_fill_discrete(na.value = "white") + 
  facet_grid(PCy ~ PCx, scales = "free") 

ggsave("outputs/1102/CRT-pca.pdf", width=14, height=12)
```


## More samples
PRJNA690579
Elwha paired end. Pulled five samples. Pulled five samples from all Campbell et al. sampling locations.

```{sh, eval=FALSE}
srun -p high -t 48:00:00 --mem=32G --nodes=2 $HOME/angsd/angsd -P 24  -bam bamlists/318.bamlist -minInd 286 -GL 1 -ref genome/omyV6Chr.fasta \
-doGLF 2 -doMajorMinor 1 -doMaf 2 -SNP_pval 1e-6 -minMapQ 20 -minQ 20 \
-out outputs/1102/318 > outputs/1102/beagle.out 2> outputs/1102/beagle.err &

python $HOME/pcangsd/pcangsd.py -beagle outputs/1102/318.beagle.gz -admix -o outputs/1102/318 -threads 10
python $HOME/pcangsd/pcangsd.py -beagle outputs/1102/318.beagle.gz -kinship -admix -admix_K 4 -o outputs/1102/admix4 -threads 10
python $HOME/pcangsd/pcangsd.py -beagle outputs/1102/318.beagle.gz -kinship -admix -admix_K 5 -o outputs/1102/admix5 -threads 10
python $HOME/pcangsd/pcangsd.py -beagle outputs/1102/318.beagle.gz -kinship -admix -admix_K 6 -o outputs/1102/admix6 -threads 10
python $HOME/pcangsd/pcangsd.py -beagle outputs/1102/318.beagle.gz -kinship -admix -admix_K 7 -o outputs/1102/admix7 -threads 10

```

Ok, I will include even more samples!!
Renaming, see 1102.1-rename.sh,
(base) maccamp@farm:~/data/trout-rad$ tr '\r' '\n' < RADmeta.csv > radmeta-unixbreaks.csv
In (base) maccamp@farm:~/data/trout-rad/Matilija$
bash ~/mccloud-rrt/1102.1-rename-bams.sh ../radmeta-unixbreaks.csv 
for f in *.sort.flt.bam; do samtools flagstat $f | grep "mapped (" | cut -f 1 -d ' ' >> stats; done;
ls | grep sort.flt.bam > bams
paste bams stats > counts.txt
cat counts.txt  | sort -k 2 -n | tail -n 5 | sort -k 1

Ventura-M017341_M165_8B_RA.sort.flt.bam	2209227
Ventura-M017354_M165_9G_RA.sort.flt.bam	1981529
Ventura-M017355_M165_9H_RA.sort.flt.bam	2243558
Ventura-M017356_M165_10A_RA.sort.flt.bam	2946142
Ventura-M017357_M165_10B_RA.sort.flt.bam	2364187

Scott-M021518_M209_2C_RA.sort.flt.bam	2674708
Scott-M021524_M209_3A_RA.sort.flt.bam	2652411
Scott-M021525_M209_3B_RA.sort.flt.bam	4472690
Scott-M021533_M209_4B_RA.sort.flt.bam	3413973
Scott-M021548_M209_6A_RA.sort.flt.bam	3734828

San-M067184_M686_3D_RA.sort.flt.bam	2145256
San-M067187_M686_3G_RA.sort.flt.bam	1912563
San-M067188_M686_3H_RA.sort.flt.bam	1804566
San-M067191_M686_4C_RA.sort.flt.bam	1867951
San-M067192_M686_4D_RA.sort.flt.bam	1637693

San-M057389_M583_3B_RA.sort.flt.bam	1571678
San-M057391_M583_3D_RA.sort.flt.bam	1454349
San-M057392_M583_3E_RA.sort.flt.bam	1439609
San-M057397_M583_4B_RA.sort.flt.bam	1000480
San-M057398_M583_4C_RA.sort.flt.bam	1564479

Populating spreadsheet:
cat counts.txt  | sort -k 2 -n | tail -n 5 | sort -k 1 | cut -f 1 

cat counts.txt  | sort -k 2 -n | tail -n 5 | sort -k 1 | cut -f 1 | perl -pe 's/_RA.sort.flt.bam//g' | cut -f 2 -d "-"

cat counts.txt  | sort -k 2 -n | tail -n 5 | sort -k 1 | cut -f 2

Creating path:
cat counts.txt  | sort -k 2 -n | tail -n 5 | sort -k 1 | cut -f 1 | while read line; do echo "/home/maccamp/data/trout-rad/Matilija/$line"; done;
cat counts.txt  | sort -k 2 -n | tail -n 5 | sort -k 1 | cut -f 1 | while read line; do echo "/home/maccamp/data/trout-rad/ScottWeir/$line"; done;
cat counts.txt  | sort -k 2 -n | tail -n 5 | sort -k 1 | cut -f 1 | while read line; do echo "/home/maccamp/data/trout-rad/WFSanGabriel/$line"; done;
cat counts.txt  | sort -k 2 -n | tail -n 5 | sort -k 1 | cut -f 1 | while read line; do echo "/home/maccamp/data/trout-rad/WFSanLuis/$line"; done;

Don't forget to index!!!
for f in *.sort.flt.bam; do samtools index $f; done;

Now we have 318 samples. 286 samples @ .9     

## Plotting

```{r}
k<-npyLoad("outputs/1102/admix4.admix.Q.npy") %>% as_tibble() 
#Need to rename all colnames smartly!

colnames(k)<-gsub("V","Q",colnames(k))
k$Bamlist<-meta$Bamlist
k<-left_join(k, meta)
k <-k %>% arrange(MajorGroup, Group, Subgroup) %>% #arrange(Q1, Q2, Q3, by_group=TRUE) %>%  ungroup() %>% mutate(Index=1:n())
  mutate(Index=1:n())

```


```{r}
q<-k %>% dplyr::select(Index, Bamlist, Q1, Q2, Q3, Q4) %>% gather(key=Ancestry, value=Q, 3:6)
qs<-left_join(q,meta)
pops2<-qs  %>% group_by(MajorGroup, Group) %>% mutate(Start=min(Index), Stop=max(Index)) %>% 
  select(Location,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2)) %>% ungroup() %>% select(-Location) %>% unique()
```


```{r}
kplot4<-ggplot(qs) +
  geom_col(aes(x=Index, y=Q, fill=Ancestry), color="NA", size = 0, width = 1)+
  geom_segment(data=pops2, x = pops2$Start - 0.5, y=0, xend = pops2$Start-0.5, yend=1, alpha=0.9, size=0.25) +
  geom_segment(data=pops2, x = pops2$Stop[length(pops2$Stop)]  + 0.5, y=0, xend= pops2$Stop[length(pops2$Stop)] + 0.5, yend=1,  alpha=0.9,
               size=0.25) +
  geom_segment(x=0, xend= pops2$Stop[length(pops2$Stop)], y=1, yend=1, alpha=0.9, size=0.25) +
  theme(axis.text.x = element_text(angle=45, hjust=1)) +
  ylim(0,1.01) +
  xlim(-0.1, pops2$Stop[length(pops2$Stop)]+1) +
  theme(panel.background = element_blank())+
  scale_x_continuous(breaks=pops2$Position, labels=pops2$Group) +
  xlab("") 
 # scale_fill_viridis_d(option="magma")
  #theme(axis.text.x = element_blank(), axis.ticks.x=element_blank()) 
#  theme(legend.position = "NA") +

kplot4
ggsave("outputs/1102/kplot4.jpeg", width=16, height=8)
```

```{r}
k<-npyLoad("outputs/1102/admix5.admix.Q.npy") %>% as_tibble() 
#Need to rename all colnames smartly!

colnames(k)<-gsub("V","Q",colnames(k))
k$Bamlist<-meta$Bamlist
k<-left_join(k, meta)
k <-k %>% arrange(MajorGroup, Group, Subgroup) %>% #arrange(Q1, Q2, Q3, by_group=TRUE) %>%  ungroup() %>% mutate(Index=1:n())
  mutate(Index=1:n())



q<-k %>% dplyr::select(Index, Bamlist, Q1, Q2, Q3, Q4, Q5) %>% gather(key=Ancestry, value=Q, 3:7)
qs<-left_join(q,meta)
pops2<-qs  %>% group_by(MajorGroup, Group) %>% mutate(Start=min(Index), Stop=max(Index)) %>% 
  select(Location,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2)) %>% ungroup() %>% select(-Location) %>% unique()

kplot5<-ggplot(qs) +
  geom_col(aes(x=Index, y=Q, fill=Ancestry), color="NA", size = 0, width = 1)+
  geom_segment(data=pops2, x = pops2$Start - 0.5, y=0, xend = pops2$Start-0.5, yend=1, alpha=0.9, size=0.25) +
  geom_segment(data=pops2, x = pops2$Stop[length(pops2$Stop)]  + 0.5, y=0, xend= pops2$Stop[length(pops2$Stop)] + 0.5, yend=1,  alpha=0.9,
               size=0.25) +
  geom_segment(x=0, xend= pops2$Stop[length(pops2$Stop)], y=1, yend=1, alpha=0.9, size=0.25) +
  theme(axis.text.x = element_text(angle=45, hjust=1)) +
  ylim(0,1.01) +
  xlim(-0.1, pops2$Stop[length(pops2$Stop)]+1) +
  theme(panel.background = element_blank())+
  scale_x_continuous(breaks=pops2$Position, labels=pops2$Group) +
  xlab("") 
 # scale_fill_viridis_d(option="magma")
  #theme(axis.text.x = element_blank(), axis.ticks.x=element_blank()) 
#  theme(legend.position = "NA") +

kplot5
ggsave("outputs/1102/kplot5.jpeg", width=16, height=8)
```

```{r}
k<-npyLoad("outputs/1102/admix6.admix.Q.npy") %>% as_tibble() 
#Need to rename all colnames smartly!

colnames(k)<-gsub("V","Q",colnames(k))
k$Bamlist<-meta$Bamlist
k<-left_join(k, meta)
k <-k %>% arrange(MajorGroup, Group, Subgroup) %>% #arrange(Q1, Q2, Q3, by_group=TRUE) %>%  ungroup() %>% mutate(Index=1:n())
  mutate(Index=1:n())



q<-k %>% dplyr::select(Index, Bamlist, Q1, Q2, Q3, Q4, Q5, Q6) %>% gather(key=Ancestry, value=Q, 3:8)
qs<-left_join(q,meta)
pops2<-qs  %>% group_by(MajorGroup, Group) %>% mutate(Start=min(Index), Stop=max(Index)) %>% 
  select(Location,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2)) %>% ungroup() %>% select(-Location) %>% unique()

kplot6<-ggplot(qs) +
  geom_col(aes(x=Index, y=Q, fill=Ancestry), color="NA", size = 0, width = 1)+
  geom_segment(data=pops2, x = pops2$Start - 0.5, y=0, xend = pops2$Start-0.5, yend=1, alpha=0.9, size=0.25) +
  geom_segment(data=pops2, x = pops2$Stop[length(pops2$Stop)]  + 0.5, y=0, xend= pops2$Stop[length(pops2$Stop)] + 0.5, yend=1,  alpha=0.9,
               size=0.25) +
  geom_segment(x=0, xend= pops2$Stop[length(pops2$Stop)], y=1, yend=1, alpha=0.9, size=0.25) +
  theme(axis.text.x = element_text(angle=45, hjust=1)) +
  ylim(0,1.01) +
  xlim(-0.1, pops2$Stop[length(pops2$Stop)]+1) +
  theme(panel.background = element_blank())+
  scale_x_continuous(breaks=pops2$Position, labels=pops2$Group) +
  xlab("") 
 # scale_fill_viridis_d(option="magma")
  #theme(axis.text.x = element_blank(), axis.ticks.x=element_blank()) 
#  theme(legend.position = "NA") +

kplot6
ggsave("outputs/1102/kplot6.jpeg", width=16, height=8)
```


```{r}
k<-npyLoad("outputs/1102/admix7.admix.Q.npy") %>% as_tibble() 
#Need to rename all colnames smartly!

colnames(k)<-gsub("V","Q",colnames(k))
k$Bamlist<-meta$Bamlist
k<-left_join(k, meta)
k <-k %>% arrange(MajorGroup,Group, Watershed) %>% #arrange(Q1, Q2, Q3, by_group=TRUE) %>%  ungroup() %>% mutate(Index=1:n())
  mutate(Index=1:n())



q<-k %>% dplyr::select(Index, Bamlist, Q1, Q2, Q3, Q4, Q5, Q6, Q7) %>% gather(key=Ancestry, value=Q, 3:9)
qs<-left_join(q,meta)
pops2<-qs  %>% group_by(MajorGroup, Group) %>% mutate(Start=min(Index), Stop=max(Index)) %>% 
  select(Location,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2)) %>% ungroup() %>% select(-Location) %>% unique()

kplot7<-ggplot(qs) +
  geom_col(aes(x=Index, y=Q, fill=Ancestry), color="NA", size = 0, width = 1)+
  geom_segment(data=pops2, x = pops2$Start - 0.5, y=0, xend = pops2$Start-0.5, yend=1, alpha=0.9, size=0.25) +
  geom_segment(data=pops2, x = pops2$Stop[length(pops2$Stop)]  + 0.5, y=0, xend= pops2$Stop[length(pops2$Stop)] + 0.5, yend=1,  alpha=0.9,
               size=0.25) +
  geom_segment(x=0, xend= pops2$Stop[length(pops2$Stop)], y=1, yend=1, alpha=0.9, size=0.25) +
  theme(axis.text.x = element_text(angle=45, hjust=1)) +
  ylim(0,1.01) +
  xlim(-0.1, pops2$Stop[length(pops2$Stop)]+1) +
  theme(panel.background = element_blank())+
  scale_x_continuous(breaks=pops2$Position, labels=pops2$Group) +
  xlab("")  
 #facet_grid(Watershed~MajorGroup, scales="free_x", space="free")

 # scale_fill_viridis_d(option="magma")
  #theme(axis.text.x = element_blank(), axis.ticks.x=element_blank()) 
#  theme(legend.position = "NA") +

kplot7
ggsave("outputs/1102/kplot7.jpeg", width=16, height=8)
```

```{r}
reds<- k %>% filter(Group %in% c("MCRT","REDB")) %>% arrange(Watershed) %>% mutate(Index=1:n()) %>%
  dplyr::select(Index, Bamlist, Q1, Q2, Q3, Q4, Q5, Q6) %>% gather(key=Ancestry, value=Q, 3:8) %>%
  left_join(meta)
pops2<-reds %>% group_by(Watershed) %>% mutate(Start=min(Index), Stop=max(Index)) %>% 
  select(Location,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2)) %>% ungroup() %>% select(-Location) %>% unique()

ggplot(reds) +
  geom_col(aes(x=Index, y=Q, fill=Ancestry), color="NA", size = 0, width = 1)+
#  geom_segment(data=pops2, x = pops2$Start - 0.5, y=0, xend = pops2$Start-0.5, yend=1, alpha=0.9, size=0.25) +
 # geom_segment(data=pops2, x = pops2$Stop[length(pops2$Stop)]  + 0.5, y=0, xend= pops2$Stop[length(pops2$Stop)] + 0.5, yend=1,  alpha=0.9,
    #           size=0.25) +
#  geom_segment(x=0, xend= pops2$Stop[length(pops2$Stop)], y=1, yend=1, alpha=0.9, size=0.25) +
  theme(axis.text.x = element_text(angle=45, hjust=1)) +
  ylim(0,1.01) +
  xlim(-0.1, pops2$Stop[length(pops2$Stop)]+1) +
  theme(panel.background = element_blank())+
  scale_x_continuous(breaks=pops2$Position, labels=pops2$Group) +
  xlab("") +
  facet_grid(Watershed~MajorGroup, scales="free_x", space="free")

ggsave("outputs/1102/redsk7.jpeg", height=10)
```

## VCF
angsd -P 4 -b bamlists/test14.bamlist -minInd 39  -out outputs/900/test14-maf05p-sites -minMaf 0.05 -minMapQ 30 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 -doGeno 4 -doPost 1 -postCutoff 0.95 -doVcf 1 -rf homoblocks/non-tetrasomic-non-inversion.txt > outputs/900/test14-make-vcf.out 2> outputs/900/test14-make-vcf.err

```{sh, eval=FALSE}
srun -p high -t 48:00:00 --mem=32G --nodes=2 $HOME/angsd/angsd -P 24  -bam outputs/1102/320.bamlist -minInd 304 -GL 1 -ref genome/omyV6Chr.fasta \
-doGLF 2 -doMajorMinor 1 -doMaf 1 -doGeno 4 -doPost 1 -SNP_pval 1e-6 -postCutoff 0.95 -doVcf 1 -minMapQ 20 -minQ 20 -r omy01: \
-out outputs/1102/320 > outputs/1102/vcf.out 2> outputs/1102/vcf.err &
```

That worked!
(base) maccamp@farm:~/mccloud-rrt/outputs/1102$ gunzip -c 320.vcf.gz | grep -v "#" | wc -l
5301

## Adding in single end fish

```{sh, eval=FALSE}
srun -p high -t 48:00:00 --mem=32G --nodes=2 $HOME/angsd/angsd -P 24  -bam bamlists/339.bamlist -minInd 305 -GL 1 -ref genome/omyV6Chr.fasta \
-doGLF 2 -doMajorMinor 1 -doMaf 1 -doGeno 4 -doPost 1 -SNP_pval 1e-6 -postCutoff 0.95 -doVcf 1 -minMapQ 20 -minQ 20 -r omy01: \
-out outputs/1102/339 > outputs/1102/339.out 2> outputs/1102/339.err &

python $HOME/pcangsd/pcangsd.py -beagle outputs/1102/339.beagle.gz -admix -o outputs/1102/339 -threads 10

```

```{r}

meta<-read_csv("meta/339.csv")
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
# Some shameless copying from Eric A.
#' @param samples character vector with the individuals IDs in the order in which
#' they were passed in the bamlist to angsd.
#' @param cov covariance matrix
covar2pcs <- function(samples, cov) {
  
  
  eig <- eigen(cov, symm = TRUE)
  PC <- as.data.frame(eig$vectors) %>%
    as_tibble() %>%
    setNames(sprintf("PC-%02d", 1:ncol(.)))
  
  samtib <- tibble(sample = samples)
  
  list(
    PCs = bind_cols(samtib, PC),
    eigevalues = eig$values
  )
}
```

```{r, warning=FALSE, message=FALSE}

cov<-read_delim("outputs/1102/339.cov", col_names=FALSE, delim=" ") %>% as.matrix()

```

```{r}
pca <- covar2pcs(meta$Bamlist, cov)

pca_long <- pca$PCs %>%
  tidyr::gather(., key = "PC", "val", -sample)

# then expand a grid of the possible comparisons (ordered)
expg <- expand.grid(sample = pca$PCs$sample,
                    PCx = sprintf("PC-%02d", 1:6),
                    PCy = sprintf("PC-%02d", 1:6),
                    stringsAsFactors = FALSE) %>%
  tibble::as_tibble()

# then left join the pca results onto that
pca_pairs <- dplyr::left_join(expg, pca_long, by = c("sample", "PCx" = "PC")) %>%
  dplyr::rename(val_x = val) %>%
  dplyr::left_join(pca_long, by = c("sample", "PCy" = "PC")) %>%
  dplyr::rename(val_y = val)

pp_meta <- pca_pairs %>%   # just keep the first 6 PCs around
  left_join(., meta, by = c("sample" = "Bamlist"))

# now, that has the first 6 PCs in it.  If we want to focus on the just the
# first 3, we could do 
npc <- 3
pp_meta2 <- pp_meta %>%
  filter( (PCx %in% sprintf("PC-%02d", 1:npc)) & 
            (PCy %in% sprintf("PC-%02d", 1:npc)) )

ggplot(pp_meta2, aes(x = val_x, y = val_y, fill = Group)) +
  geom_point(pch = 21, size = 2) +
  scale_fill_discrete(na.value = "white") + 
  facet_grid(PCy ~ PCx, scales = "free")


```