---
title: "300-omy05-PCA"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

#Omy05 PCA
I want to know what the omy05 region looks like in a PCA for our data sets. Starting with test data set 04, reduced to RR types and omitting LCT. Note: subsetting to omy05:site-site did require indexed bams.

```{sh, eval=FALSE}
angsd -P 4 -b bamlists/test4-omy05RR.bamlist -minInd 35 -ref genome/omyV6Chr.fasta -out outputs/300/test4-omy05 -GL 1 -doGlf 2 -doMajorMinor 1 -doMaf 2 -SNP_pval 1e-6 -minMapQ 30 -minQ 20 -r  omy05:27001895-81791946

#****ing Zcat issue
#300 mac$ ln -s test4-omy05.beagle.gz test4-omy05.beagle.Z
python /Users/mac/pcangsd/pcangsd.py  -beagle outputs/300/test4-omy05.beagle.Z -admix -o outputs/300/test4-omy05

```

## Some work to actually plot the PCA

```{r, echo=FALSE, warning=FALSE, message=FALSE}
# Some shameless copying from Eric A.
#' @param samples character vector with the individuals IDs in the order in which
#' they were passed in the bamlist to angsd.
#' @param cov covariance matrix
covar2pcs <- function(samples, cov) {
  
  
  eig <- eigen(cov, symm = TRUE)
  PC <- as.data.frame(eig$vectors) %>%
    as_tibble() %>%
    setNames(sprintf("PC-%02d", 1:ncol(.)))
  
  samtib <- tibble(sample = samples)
  
  list(
    PCs = bind_cols(samtib, PC),
    eigevalues = eig$values
  )
}
```

```{r, warning=FALSE, message=FALSE}
library(tidyverse)
pca_meta <- read_tsv("bamlists/test4-omy05RR.bamlist", col_names = FALSE) %>% mutate(Bam=gsub("bams/","",X1)) %>% 
  mutate(Sample=gsub("bams/","",Bam))
pca_meta$Sample=gsub("_0\\d_RA.sort.flt.bam","",pca_meta$Sample)
pca_meta<-pca_meta %>% mutate(Taxon=gsub("_\\w\\w\\w\\w", "", Sample)) %>% select(-X1)

# Now to get things with taxon more reasonable
pca_meta$Taxon<-ifelse(pca_meta$Sample=="RAIN_EGLK","EGLK", pca_meta$Taxon)
pca_meta$Taxon<-ifelse(pca_meta$Sample %in% c("REDB_SHPN","REDB_EDSN","REDB_MOHD"),"MRRT", pca_meta$Taxon)

cov<-read_tsv("outputs/300/test4-omy05.cov", col_names=FALSE) %>% as.matrix()
pca <- covar2pcs(pca_meta$Bam, cov)

pca_long <- pca$PCs %>%
  tidyr::gather(., key = "PC", "val", -sample)

# then expand a grid of the possible comparisons (ordered)
expg <- expand.grid(sample = pca$PCs$sample,
                    PCx = sprintf("PC-%02d", 1:6),
                    PCy = sprintf("PC-%02d", 1:6),
                    stringsAsFactors = FALSE) %>%
  tibble::as_tibble()

# then left join the pca results onto that
pca_pairs <- dplyr::left_join(expg, pca_long, by = c("sample", "PCx" = "PC")) %>%
  dplyr::rename(val_x = val) %>%
  dplyr::left_join(pca_long, by = c("sample", "PCy" = "PC")) %>%
  dplyr::rename(val_y = val)

pp_meta <- pca_pairs %>%   # just keep the first 6 PCs around
  left_join(., pca_meta, by = c("sample" = "Bam"))

# now, that has the first 6 PCs in it.  If we want to focus on the just the
# first 3, we could do 
npc <- 3
pp_meta2 <- pp_meta %>%
  filter( (PCx %in% sprintf("PC-%02d", 1:npc)) & 
            (PCy %in% sprintf("PC-%02d", 1:npc)) )

ggplot(pp_meta2, aes(x = val_x, y = val_y, fill = Taxon)) +
  geom_point(pch = 21, size = 2) +
  scale_fill_discrete(na.value = "white") + 
  facet_grid(PCy ~ PCx, scales = "free")


```
  

```{r, warning=FALSE, message=FALSE}
library(ggrepel)
pp_meta12 <- pp_meta %>%
  filter(PCx==("PC-01")) %>%
  filter(PCy==("PC-02"))

centers12<-pp_meta12 %>% group_by(Taxon) %>% mutate(MeanX=mean(val_x), MeanY=mean(val_y)) %>% select(Taxon, MeanX, MeanY) %>% unique()

#Get proportion of variance
eig <- eigen(cov, symm = TRUE)
var<-eig$values/sum(eig$values)

pc12<-ggplot()+
  geom_point(data=pp_meta12, aes(x = val_x, y = val_y, color=Taxon), size=2, alpha=0.75)+
  geom_text_repel(data=centers12, aes(x=MeanX, y=MeanY, label=Taxon), fontface='bold', size=3)+
  scale_fill_brewer(palette="Set1")+
  theme_bw()+
  theme(panel.grid=element_blank())+
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = ""))+
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = ""))+
  theme(legend.position = "")

pc12
```

```{r, warning=FALSE, message=FALSE}
library(ggrepel)
pp_meta13 <- pp_meta %>%
  filter(PCx==("PC-01")) %>%
  filter(PCy==("PC-03"))

centers13<-pp_meta13 %>% group_by(Taxon) %>% mutate(MeanX=mean(val_x), MeanY=mean(val_y)) %>% select(Taxon, MeanX, MeanY) %>% unique()

#Get proportion of variance
eig <- eigen(cov, symm = TRUE)
var<-eig$values/sum(eig$values)

pc13<-ggplot()+
  geom_point(data=pp_meta13, aes(x = val_x, y = val_y, color=Taxon), size=2, alpha=0.75)+
  geom_text_repel(data=centers13, aes(x=MeanX, y=MeanY, label=Taxon), fontface='bold', size=3)+
  scale_fill_brewer(palette="Set1")+
  theme_bw()+
  theme(panel.grid=element_blank())+
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = ""))+
  ylab(paste("PC3", " ", round((100*var[3]),2), "%", sep = ""))+
  theme(legend.position = "")

pc13
```

How much variance is in our PCs? That is, why did I display the first 3?


```{r, warning=FALSE, message=FALSE}
round(var[1:5]*100,2)
```

#This all looks great. But, what about a tree?

```{sh, eval=FALSE}

angsd -P 4 -b bamlists/test4-omy05RR.bamlist -minInd 35 -ref genome/omyV6Chr.fasta -out outputs/300/test4-omy05-genotypes -doGeno 4 -postCutoff 0.95 -doPost 1 -GL 1 -doGlf 2 -doMajorMinor 1 -doMaf 1 -minMaf 0.05 -SNP_pval 1e-6 -minMapQ 30 -minQ 20 -r omy05:27001895-81791946

./102.1-convert-omy05-to-phylip.R outputs/300/ test4-omy05-genotypes.geno.gz ./bamlists/test4-omy05RR.bamlist

source activate py3; ./103-remove-invariant.py -p outputs/300/test4-omy05-genotypes-omy05.phy -o outputs/300/test4-omy05-genotypes-omy05.asc.phy; source deactivate

#203 sites~

iqtree -s outputs/300/test4-omy05-genotypes-omy05.asc.phy -st DNA -m GTR+G4+F+ASC -bb 1000 -alrt 1000
```

Drum roll .....   


```{r, warning=FALSE, message=FALSE}
library(ggtree)
library(ape)
tree<-read.tree(file="outputs/300/test4-omy05-genotypes-omy05.asc.phy.contree")
ggtree(tree, layout="equal_angle")+geom_tiplab(size=2)+geom_nodelab()

```

## What about a network?
Can try phangorn: https://cran.r-project.org/web/packages/phangorn/vignettes/Networx.html

```{r, warning=FALSE, message=FALSE}
library(phangorn)
dat<-read.dna("outputs/300/test4-omy05-genotypes-omy05.asc.phy", format="sequential")
dat<-as.phyDat(dat)
set.seed(1)
bs <- bootstrap.phyDat(dat, FUN = function(x)nj(dist.hamming(x)), 
    bs=100)
tree <- nj(dist.hamming(dat))
par("mar" = rep(1, 4))
tree <- plotBS(tree, bs, "phylogram")

cnet <- consensusNet(bs, .3)
plot(cnet, "2D", show.edge.label=TRUE)
```
