---
title: "700-expanded-phylogeny"
author: "Mac Campbell"
date: "1/27/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# This Document is to Demonstrate Two Ideas:
__1__ Concatenated Phylogeny with Expanded MRRT Sampling
__2__ Admixture plot of Eagle Lake omy05 genotypes with MRRT and HRNB

## 1. Concatenated Phylogeny
I have included more MRRT samples here and have chosen the following samples for this test:
CAGT_VCLS_01    
CAGT_VCLS_02    
CAGT_STLC_01    
CAGT_STLC_02    
LKGT_UWLC_01    
LKGT_UWLC_02    
LKGT_SILV_01    
LKGT_SILV_02    
KRNB_CHGC_01    
KRNB_KKWC_01    
REDB_CTWD_01    
REDB_CTWD_02    
REDB_DISM_01    
REDB_DISM_02    
REDB_SHLD_01    
REDB_SHLD_02    
REDB_FTZH_02    
REDB_FTZH_05    
RAIN_EGLK_01    
RAIN_EGLK_02    
STLH_LYBA_01    
STLH_LYBA_02    
HRNB_COLE_01    
HRNB_COLE_02    
REDB_SHPN_01    
REDB_SHPN_02    
REDB_EDSN_01    
REDB_EDSN_04    
REDB_MOHD_01    
REDB_MOHD_02    
LCT_GACR_10    
PCT_PPP1_01    
    

Now we have a test6.bamlist and test6-sample-names.txt. We can run this locally, using ~ 95% missing data threshold. I am excluding tetrasomically inhertied genomic regions and the inversions on omy05 and omy20.    

```{sh, eval=FALSE}
angsd -P 4 -b bamlists/test6.bamlist -minInd 30  -out outputs/700/test6-maf05p-sites -minMaf 0.05 -minMapQ 30 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 -doGeno 4 -doPost 1 -postCutoff 0.95 -doVcf 1 -rf homoblocks/non-tetrasomic-non-inversion.txt > outputs/700/make-vcf.out 2> outputs/700/make-vcf.err

#Sort vcf
gunzip -c outputs/700/test6-maf05p-sites.vcf.gz | awk '$1 ~ /^#/ {print $0;next} {print $0 | "sort -k1,1 -k2,2n"}' > outputs/700/test6-maf05p-sites.sort.vcf
#Compress
bgzip outputs/700/test6-maf05p-sites.sort.vcf 
#Index
tabix outputs/700/test6-maf05p-sites.sort.vcf.gz -f
#Prune
bcftools +prune -l 0.9 -w 30000 outputs/700/test6-maf05p-sites.sort.vcf.gz -Ov -o outputs/700/test6-maf05p-sites.sort.prune.vcf

#lets reheader the bam file
bcftools reheader --samples bamlists/test6-sample-names.txt -o outputs/700/test6-maf05p-sites.sort.prune.reheadered.vcf outputs/700/test6-maf05p-sites.sort.prune.vcf

#Convert to phylip
./vcf2phylip.py -i outputs/700/test6-maf05p-sites.sort.prune.reheadered.vcf

#Correct for ascertainment bias
source activate py3; ./103-remove-invariant.py -p outputs/700/test6-maf05p-sites.sort.prune.reheadered.min4.phy -o outputs/700/test6-maf05p-sites.sort.prune.reheadered.min4.asc.phy; source deactivate;

#Infer a proper tree
iqtree -s outputs/700/test6-maf05p-sites.sort.prune.reheadered.min4.asc.phy -st DNA -m GTR+G4+F+ASC -bb 1000 -alrt 1000 --redo

```
Note:Alignment has 32 sequences with 5917 columns, 5899 distinct patterns
3408 parsimony-informative, 2509 singleton sites, 0 constant sites


Plot this tree.    
```{r, warning=FALSE, message=FALSE}
library(tidyverse)
library(ggtree)
library(ape)
tree<-read.tree(file="outputs/700/test6-maf05p-sites.sort.prune.reheadered.min4.asc.phy.contree")
tree<-root(tree, c("PCT_PPP1_01","LCT_GACR_10"))

ggtree(tree)+geom_tiplab(size=2)+geom_nodelab()


```
How about a distance tree? 

```{r, warning=FALSE, message=FALSE}
library(phangorn)
dna<-read.phyDat("outputs/700/test6-maf05p-sites.sort.prune.reheadered.min4.asc.phy")
dat<-as.phyDat(dna)
set.seed(1)
bs <- bootstrap.phyDat(dat, FUN = function(x)nj(dist.hamming(x)), 
    bs=100)
tree <- nj(dist.hamming(dat))
par("mar" = rep(1, 4))
tree <- plotBS(tree, bs, "phylogram")

#By default prob=0.3
#cnet <- consensusNet(bs, .25)
#plot(cnet, "2D", show.edge.label=TRUE)

```


It seems to me that the inclusion of LCT has a substantial impact on the phylogeny, the negative branch lengths and attachment point are bizzarre.  Test7 lacks LCT.   
```{sh, eval=FALSE}
angsd -P 4 -b bamlists/test7.bamlist -minInd 29  -out outputs/700/test7-maf05p-sites -minMaf 0.05 -minMapQ 30 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 -doGeno 4 -doPost 1 -postCutoff 0.95 -doVcf 1 -rf homoblocks/non-tetrasomic-non-inversion.txt > outputs/700/test7-make-vcf.out 2> outputs/700/test7-make-vcf.err

#Sort vcf
gunzip -c outputs/700/test7-maf05p-sites.vcf.gz | awk '$1 ~ /^#/ {print $0;next} {print $0 | "sort -k1,1 -k2,2n"}' > outputs/700/test7-maf05p-sites.sort.vcf
#Compress
bgzip outputs/700/test7-maf05p-sites.sort.vcf 
#Index
tabix outputs/700/test7-maf05p-sites.sort.vcf.gz -f
#Prune
bcftools +prune -l 0.9 -w 30000 outputs/700/test7-maf05p-sites.sort.vcf.gz -Ov -o outputs/700/test7-maf05p-sites.sort.prune.vcf

#lets reheader the bam file
bcftools reheader --samples bamlists/test7-sample-names.txt -o outputs/700/test7-maf05p-sites.sort.prune.reheadered.vcf outputs/700/test7-maf05p-sites.sort.prune.vcf

#Convert to phylip
./vcf2phylip.py -i outputs/700/test7-maf05p-sites.sort.prune.reheadered.vcf

#Correct for ascertainment bias
source activate py3; ./103-remove-invariant.py -p outputs/700/test7-maf05p-sites.sort.prune.reheadered.min4.phy -o outputs/700/test7-maf05p-sites.sort.prune.reheadered.min4.asc.phy; source deactivate;

#Infer a proper tree
iqtree -s outputs/700/test7-maf05p-sites.sort.prune.reheadered.min4.asc.phy -st DNA -m GTR+G4+F+ASC -bb 1000 -alrt 1000

```

#Plot this tree
```{r, warning=FALSE, message=FALSE}
tree2<-read.tree(file="outputs/700/test7-maf05p-sites.sort.prune.reheadered.min4.asc.phy.contree")

ggtree(tree2)+geom_tiplab(size=2)+geom_nodelab()
```

```{r, warning=FALSE, message=FALSE}
library(phangorn)
dna<-read.phyDat("outputs/700/test7-maf05p-sites.sort.prune.reheadered.min4.asc.phy")
dat<-as.phyDat(dna)
set.seed(1)
bs <- bootstrap.phyDat(dat, FUN = function(x)nj(dist.hamming(x)), 
    bs=100)
tree <- nj(dist.hamming(dat))
par("mar" = rep(1, 4))
tree <- plotBS(tree, bs, "phylogram")

#By default prob=0.3
cnet <- consensusNet(bs, .25)
plot(cnet, "2D", show.edge.label=TRUE)

```


I can consider the Goose Lake and Warner Lakes Redbands outgroups, or at least one of them, and do some other analyses.
```{sh, eval=FALSE}
#Generate treemix files, first edit header of vcf, 26002 out of 26075 sites kept
./600.1-vcf2treemix.sh outputs/700/test7-maf05p-sites.sort.prune.vcf  outputs/700/test7.clst


for i in {0..5}
do
 treemix -i outputs/700/test7-maf05p-sites.sort.prune.treemix.frq.gz -m $i -o outputs/700/test7.$i -root WLRT -bootstrap -k 500  > outputs/700/test7.treemix_${i}_log &
done
```

Plotting
```{r, warning=FALSE,message=FALSE}
library(RColorBrewer)
library(R.utils)
source("~/treemix/treemix-1.13/src/plotting_funcs.R")
prefix="outputs/700/test7"

par(mfrow=c(2,3))
for(edge in 0:5){
  plot_tree(cex=0.8,paste0(prefix,".",edge))
  title(paste(edge,"edges"))
}
```
