---
title: "901-CRRB"
author: "Mac Campbell"
date: "3/10/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

#O. m. gairdneri
Does single-end RADseq help us out here?     
Created a test14.bamlist    
```{sh, eval=FALSE}
angsd -P 4 -b bamlists/test14.bamlist -minInd 39  -out outputs/900/test14-maf05p-sites -minMaf 0.05 -minMapQ 30 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 -doGeno 4 -doPost 1 -postCutoff 0.95 -doVcf 1 -rf homoblocks/non-tetrasomic-non-inversion.txt > outputs/900/test14-make-vcf.out 2> outputs/900/test14-make-vcf.err

#Sort vcf
gunzip -c outputs/900/test14-maf05p-sites.vcf.gz | awk '$1 ~ /^#/ {print $0;next} {print $0 | "sort -k1,1 -k2,2n"}' > outputs/900/test14-maf05p-sites.sort.vcf
#Compress
bgzip outputs/900/test14-maf05p-sites.sort.vcf 
#Index
tabix outputs/900/test14-maf05p-sites.sort.vcf.gz -f

#Pruned - as input for multispecies coalescent type analyses
bcftools +prune -l 0.9 -w 30000 outputs/900/test14-maf05p-sites.sort.vcf.gz -Ov -o outputs/900/test14-maf05p-sites.sort.prune.vcf

#lets reheader the bam file
bcftools reheader --samples bamlists/test14-sample-names.txt -o outputs/900/test14-maf05p-sites.sort.prune.reheadered.vcf outputs/900/test14-maf05p-sites.sort.prune.vcf

#Convert to phylip
./vcf2phylip.py -i outputs/900/test14-maf05p-sites.sort.prune.reheadered.vcf

#Correct for ascertainment bias
source activate py3; ./103-remove-invariant.py -p outputs/900/test14-maf05p-sites.sort.prune.reheadered.min4.phy -o outputs/900/test14-maf05p-sites.sort.prune.reheadered.min4.asc.phy; source deactivate;

#Infer a proper tree, with model testing
iqtree -s outputs/900/test14-maf05p-sites.sort.prune.reheadered.min4.asc.phy -st DNA -m MFP+ASC -bb 1000 -alrt 1000 --redo

#Unpruned
bcftools reheader --samples bamlists/test14-sample-names.txt -o outputs/900/test14-maf05p-sites.sort.reheadered.vcf.gz outputs/900/test14-maf05p-sites.sort.vcf.gz

gunzip outputs/900/test14-maf05p-sites.sort.reheadered.vcf.gz

./vcf2phylip.py -i outputs/900/test14-maf05p-sites.sort.reheadered.vcf

source activate py3; ./103-remove-invariant.py -p outputs/900/test14-maf05p-sites.sort.reheadered.min4.phy -o outputs/900/test14-maf05p-sites.sort.reheadered.min4.asc.phy; source deactivate;

iqtree -s outputs/900/test14-maf05p-sites.sort.reheadered.min4.asc.phy -st DNA -m MFP+ASC  -bb 1000 -alrt 1000 --redo



```

I end up with 41 taxa and 2199 unlinked SNPs.

How do our trees look?    

```{r, warning=FALSE, message=FALSE}
library(tidyverse)
library(ggtree)
library(ape)

tree<-read.tree(file="outputs/900/test14-maf05p-sites.sort.prune.reheadered.min4.asc.phy.contree")
tree<-root(tree, c("PCT_PPP1_01","LCT_GACR_10"))
ggtree(tree)+geom_tiplab(size=2)+geom_nodelab()

tree2<-read.tree(file="outputs/900/test14-maf05p-sites.sort.reheadered.min4.asc.phy.contree")
tree2<-root(tree, c("PCT_PPP1_01","LCT_GACR_10"))
ggtree(tree2)+geom_tiplab(size=2)+geom_nodelab()


```


## Species Tree

```{r, warning=FALSE, message=FALSE, out.width='75%'}
t2<-read.nexus(file="outputs/900/svdq/svdq-test.asc.tre")
t2<-root(t2, c("LCT"))
ggtree(t2,branch.length = "none")+geom_tiplab(size=4)+geom_nodelab()+xlim(0,8)
ggsave("outputs/900/svdq/svdq-test.asc.jpg", width=7, height=5)
```

For consistency, I can split this by sampling location.
```{r, warning=FALSE, message=FALSE, out.width='75%'}
t2<-read.nexus(file="outputs/900/svdq/sampling-locations-test.asc.tre")
t2<-root(t2, c("LCT"))
ggtree(t2,branch.length = "none")+geom_tiplab(size=3)+geom_nodelab()+xlim(0,9)
ggsave("outputs/900/svdq/svdq-test.asc.jpg", width=7, height=5)
```

#Network
I'll make a nexus file for analysis in splitstree.
```{r, eval=FALSE, warning=FALSE, message=FALSE}
library(treeio)
data<-read.phylip.seq("outputs/900/test14-maf05p-sites.sort.prune.reheadered.min4.asc.phy")
write.nexus.data(data, file="outputs/900/splits.nex")
```


```{r nnet, echo=FALSE, fig.cap="Neighbor-Net", out.width = '100%'}
knitr::include_graphics("outputs/900/splits.jpg")
```