---
title: "906-more-cutties"
output: html_document
date: "2023-04-25"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

```{r}
library(tidyverse)
library(ggtree)
library(ape)
```

## More Cutties

I rounded up some YCT and WCT

```{sh, eval=FALSE}
angsd -P 4 -b bamlists/test-cutties.bamlist -minInd 45  -out outputs/906/cutties-maf05p-sites -minMaf 0.05 -minMapQ 10 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 -doGeno 4 -doPost 1 -postCutoff 0.95 -doVcf 1 -rf homoblocks/non-tetrasomic-non-inversion.txt > outputs/906/cutties-make-vcf.out 2> outputs/906/cutties-make-vcf.err

angsd -P 4 -b bamlists/test-cutties.bamlist -minInd 43  -out outputs/906/cutties-maf05p-sites-min43 -minMaf 0.05 -minMapQ 10 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 -doGeno 4 -doPost 1 -postCutoff 0.95 -doVcf 1 -rf homoblocks/non-tetrasomic-non-inversion.txt > outputs/906/cutties-make-vcf.out 2> outputs/906/cutties-make-vcf.err

#Sort vcf
gunzip -c outputs/906/cutties-maf05p-sites.vcf.gz | awk '$1 ~ /^#/ {print $0;next} {print $0 | "sort -k1,1 -k2,2n"}' > outputs/906/cutties-maf05p-sites.sort.vcf

#Compress
bgzip outputs/906/cutties-maf05p-sites.sort.vcf 
#Index
tabix outputs/906/cutties-maf05p-sites.sort.vcf.gz -f

#Pruned - as input for multispecies coalescent type analyses
bcftools +prune -l 0.5 -w 10000 outputs/906/cutties-maf05p-sites.sort.vcf.gz -Ov -o outputs/906/cutties-maf05p-sites.sort.prune.vcf

#lets reheader the bam file
bcftools reheader --samples bamlists/test-cutties-sample-names.txt -o outputs/906/cutties-maf05p-sites.sort.prune.reheadered.vcf outputs/906/cutties-maf05p-sites.sort.prune.vcf

#Convert to phylip
./vcf2phylip.py -i outputs/906/cutties-maf05p-sites.sort.prune.reheadered.vcf
```


Test tree

```{sh, eval=FALSE}
iqtree -s outputs/906/cutties-maf05p-sites.sort.prune.reheadered.min4.phy -st DNA -m MFP+ASC -bb 1000 -alrt 1000 --redo
iqtree -s outputs/906/cutties-maf05p-sites.sort.prune.reheadered.min4.phy.varsites.phy -st DNA -m MFP+ASC -bb 1000 -alrt 1000 --redo

```
Alignment has 47 sequences with 1620 columns, 1615 distinct patterns

```{r}
data<-read.dna("outputs/906/cutties-maf05p-sites.sort.prune.reheadered.min4.phy.varsites.phy")
write.nexus.data(data, file="outputs/906/splits.nex")
```


```{r nnet, echo=FALSE, fig.cap="Neighbor-Net", out.width = '100%'}
knitr::include_graphics("outputs/900/splits.jpg")
```
