---
title: "population-genetics-forward-only"
author: "Mac Campbell"
date: "9/21/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

```{r}
library(tidyverse)
library(viridis)
library(ggpubr)
```

## Forward Only

I have some forward only Samples. Can I get them to play nice?

```{sh, eval=FALSE}
for f in *.1; do fastq-dump $f; done;
run_align_single.sh seqs ~/mccloud-rrt/genome/omyV6Chr.fasta
```


```{r}
meta<-read_csv("meta/combined.csv")
```

We can take the forward only reads from "REDB_MILL_02_RA.sort.bam" and they should be compatible.

  first in pair (0x40)    

Putting in /home/maccamp/mccloud-rrt/bams/forward    

```{sh, eval=FALSE}
samtools view -f 0x40 -b /home/ehabibi/projects/redbandnewalign/bamfiles2/REDB_MILL_02_RA.sort.bam > REDB_MILL_02_RA.fwd.bam
```

This does look to be what we want.... 

```{sh, eval=FALSE}
head bamlists/318.bamlist | while read line; do samtools view -f 0x40 -b $line > bams/forward/`basename $line .sort.flt.bam`.fwd.bam ; done;
```

The coverage of the genbank samples looks to be much higher overall.
samtools flagstat ../SRR5933417.1.fastq.sort.bam
2236738 + 0 in total (QC-passed reads + QC-failed reads)

Loop takes a bit to run, but I think that I can wait a second while it runs.

```{sh, eval=FALSE}
srun -p high -t 4:00:00 cat bamlists/318.bamlist | while read line; do samtools view -b -f 0x40 -b $line > bams/forward/`basename $line .sort.flt.bam`.fwd.bam ; done;
```

Ok, now we have 318 samples in bams/forward like *.fwd.bam

Problem with with no sam header:
samtools view -b -f 0x40 ~/data/trout-rad/WFSanLuis/San-M057398_M583_4C_RA.sort.flt.bam -t genome/omyV6Chr.fasta.fai > bams/forward/San-M057398_M583_4C_RA.fwd.bam 

```{sh, eval=FALSE}
for f in *.fwd.bam; do samtools index $f; done;
```


```{r}
bam318<-read_tsv("bamlists/318.bamlist", col_names = c("Path"))
bam318$Path<-gsub("sourced","forward", bam318$Path)
bam318$Path<-gsub("/home/maccamp/data/trout-rad/ScottWeir/|/home/maccamp/data/trout-rad/BigCk|/home/maccamp/data/elwha/|/home/maccamp/data/trout-rad/Matilija/|/home/maccamp/data/trout-rad/WFSanGabriel/|/home/maccamp/data/trout-rad/WFSanLuis/", "bams/forward/",bam318$Path)

bam318$Path<-gsub("sort.flt.bam","fwd.bam",bam318$Path)

bamsingle<-filter(meta, Type=="Single") %>% select(Bamlist) %>% rename(Path=Bamlist)
##55 in bamssingle

bams<-bind_rows(bam318,bamsingle)


write_tsv(bams, "bamlists/373.single.bamlist", col_names = FALSE)
```

```{r}
meta$Singles<-bams$Path
```



## PCA

First using only omy01

```{sh, eval=FALSE}
srun -p high -t 48:00:00 --mem=32G --nodes=1 $HOME/angsd/angsd -P 12  -bam bamlists/373.single.bamlist -minInd 336 -GL 1 -ref genome/omyV6Chr.fasta \
-doGLF 2 -doMajorMinor 1 -doMaf 2 -SNP_pval 1e-6 -minMapQ 20 -minQ 20 -r omy01: \
-out outputs/population-genetics-forward/373.single > outputs/population-genetics-forward/beagle.out 2> outputs/population-genetics-forward/beagle.err &

srun -p high -t 1:00:00 python $HOME/pcangsd/pcangsd.py -beagle outputs/population-genetics-forward/373.single.beagle.gz -o outputs/population-genetics-forward/373.single -threads 10
```

339:Number of sites after MAF filtering (0.05): 1915 for omy01

373: 

Maybe we should calculate average coverage and also consider raising MAF? Let's see how it plays out.

### PCA

```{r, echo=FALSE, warning=FALSE, message=FALSE}
# Some shameless copying from Eric A.
#' @param samples character vector with the individuals IDs in the order in which
#' they were passed in the bamlist to angsd.
#' @param cov covariance matrix
covar2pcs <- function(samples, cov) {
  
  
  eig <- eigen(cov, symm = TRUE)
  PC <- as.data.frame(eig$vectors) %>%
    as_tibble() %>%
    setNames(sprintf("PC-%02d", 1:ncol(.)))
  
  samtib <- tibble(sample = samples)
  
  list(
    PCs = bind_cols(samtib, PC),
    eigevalues = eig$values
  )
}
```

```{r, warning=FALSE, message=FALSE}
cov<-read_delim("outputs/population-genetics-single/373.single.cov", col_names=FALSE, delim=" ") %>% as.matrix()
```


```{r}
pca <- covar2pcs(meta$Bamlist, cov)

pca_long <- pca$PCs %>%
  tidyr::gather(., key = "PC", "val", -sample)

# then expand a grid of the possible comparisons (ordered)
expg <- expand.grid(sample = pca$PCs$sample,
                    PCx = sprintf("PC-%02d", 1:6),
                    PCy = sprintf("PC-%02d", 1:6),
                    stringsAsFactors = FALSE) %>%
  tibble::as_tibble()

# then left join the pca results onto that
pca_pairs <- dplyr::left_join(expg, pca_long, by = c("sample", "PCx" = "PC")) %>%
  dplyr::rename(val_x = val) %>%
  dplyr::left_join(pca_long, by = c("sample", "PCy" = "PC")) %>%
  dplyr::rename(val_y = val)

pp_meta <- pca_pairs %>%   # just keep the first 6 PCs around
  left_join(., meta, by = c("sample" = "Bamlist"))

# now, that has the first 6 PCs in it.  If we want to focus on the just the
# first 3, we could do 
npc <- 3
pp_meta2 <- pp_meta %>%
  filter( (PCx %in% sprintf("PC-%02d", 1:npc)) & 
            (PCy %in% sprintf("PC-%02d", 1:npc)) )

ggplot(pp_meta2, aes(x = val_x, y = val_y, fill = Group)) +
  geom_point(pch = 21, size = 2) +
  scale_fill_discrete(na.value = "white") + 
  facet_grid(PCy ~ PCx, scales = "free")


```
```{r}

eig <- eigen(cov, symm = TRUE)
var<-eig$values/sum(eig$values)
cumvar<-cumsum(eig$values)/sum(eig$values)

head(var)
head(cumvar)
```


```{r}
sub12<-pp_meta2 %>% filter( (PCx =="PC-01") & (PCy =="PC-02") )

pc12<-ggplot(sub12, aes(x = val_x, y = val_y, fill = Group, shape=MajorGroup)) +
  geom_point(size = 2, alpha=0.75) +
  scale_fill_discrete(na.value = "white") + 
  theme_bw()+
  theme(panel.grid=element_blank())+
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = ""))+
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = ""))+
  scale_fill_viridis_d(option = "magma") +
  scale_shape_manual("Major Group", values = c(21,22,22,23,24,25)) +
  theme(legend.position = "")+
  ggtitle("A") +
  theme(plot.title = element_text(face="bold", size=16))

sub13<-pp_meta2 %>% filter( (PCx =="PC-01") & (PCy =="PC-03") )

pc13<-ggplot(sub13, aes(x = val_x, y = val_y, fill = Group, shape=MajorGroup)) +
  geom_point(size = 2, alpha=0.75) +
  scale_fill_discrete(na.value = "white") + 
  theme_bw()+
  theme(panel.grid=element_blank())+
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = ""))+
  ylab(paste("PC3", " ", round((100*var[3]),2), "%", sep = ""))+
  scale_fill_viridis_d(option = "magma") +
  scale_shape_manual("Major Group", values = c(21,22,22,23,24,25)) +
  guides(fill = guide_legend(override.aes=list(shape=15, color= viridis(8, option="magma")))) +
  ggtitle("B") +
  theme(plot.title = element_text(face="bold", size=16))


pcs<-ggarrange(pc12, pc13, ncol = 2, widths=c(1.1, 1.4))
pcs
```

well, it is similar.

```{r}
pc12 + ggtitle("PCA of Forward Only Data") + theme(plot.title = element_text(hjust=0.5)) +
  guides(fill = guide_legend(override.aes=list(shape=15, color= viridis(8, option="magma")))) +
  theme(legend.position = "right")
ggsave("outputs/population-genetics-single/373.PC12.jpeg")
```

Looking at redb
```{r}

ggplot(sub12 %>% filter(MajorGroup %in% c("REDB","MCRT","CRT")), aes(x = val_x, y = val_y, fill = Group, shape=MajorGroup)) +
  geom_point(size = 2, alpha=0.75) +
  scale_fill_discrete(na.value = "white") + 
  theme_bw()+
  theme(panel.grid=element_blank())+
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = ""))+
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = ""))+
  scale_shape_manual("Major Group", values = c(21,22,22,23,24,25)) +
  theme(plot.title = element_text(face="bold", size=16)) +
  guides(fill = guide_legend(override.aes=list(shape=22)))


```

Subsetting:

```{r}
redb<-meta %>% filter(OriginalGroup=="REDB")

write_tsv(select(redb, Singles), "bamlists/259.bamlist",col_names = FALSE)
```

```{sh, eval=FALSE}
srun -p high -t 48:00:00 --mem=32G --nodes=1 $HOME/angsd/angsd -P 12  -bam bamlists/259.bamlist -minInd 233 -GL 1 -ref genome/omyV6Chr.fasta \
-doGLF 2 -doMajorMinor 1 -doMaf 2 -SNP_pval 1e-6 -minMapQ 20 -minQ 20 -r omy01: \
-out outputs/population-genetics-forward/259.single > outputs/population-genetics-forward/259beagle.out 2> outputs/population-genetics-forward/259beagle.err &

srun -p high -t 1:00:00 python $HOME/pcangsd/pcangsd.py -beagle outputs/population-genetics-forward/259.single.beagle.gz -o outputs/population-genetics-forward/259.single -threads 10
```

225: Number of sites after MAF filtering (0.05): 1322
```{r, warning=FALSE, message=FALSE}

cov<-read_delim("outputs/population-genetics-single/259.single.cov", col_names=FALSE, delim=" ") %>% as.matrix()

```

```{r}
pca <- covar2pcs(redb$Bamlist, cov)

pca_long <- pca$PCs %>%
  tidyr::gather(., key = "PC", "val", -sample)

# then expand a grid of the possible comparisons (ordered)
expg <- expand.grid(sample = pca$PCs$sample,
                    PCx = sprintf("PC-%02d", 1:6),
                    PCy = sprintf("PC-%02d", 1:6),
                    stringsAsFactors = FALSE) %>%
  tibble::as_tibble()

# then left join the pca results onto that
pca_pairs <- dplyr::left_join(expg, pca_long, by = c("sample", "PCx" = "PC")) %>%
  dplyr::rename(val_x = val) %>%
  dplyr::left_join(pca_long, by = c("sample", "PCy" = "PC")) %>%
  dplyr::rename(val_y = val)

pp_meta <- pca_pairs %>%   # just keep the first 6 PCs around
  left_join(., redb, by = c("sample" = "Bamlist"))

# now, that has the first 6 PCs in it.  If we want to focus on the just the
# first 3, we could do 
npc <- 4
pp_meta2 <- pp_meta %>%
  filter( (PCx %in% sprintf("PC-%02d", 1:npc)) & 
            (PCy %in% sprintf("PC-%02d", 1:npc)) )

ggplot(pp_meta2) +
  geom_point(aes(x=val_x, y=val_y, fill=Watershed, shape=MajorGroup), alpha=0.75) +
  theme_bw() +
  scale_shape_manual(values = c(21,24)) +
  facet_grid(PCx ~ PCy) +
  guides(fill = guide_legend(override.aes=list(shape=22)))

```


```{r}

eig <- eigen(cov, symm = TRUE)
var<-eig$values/sum(eig$values)
cumvar<-cumsum(eig$values)/sum(eig$values)

head(var)
head(cumvar)
```


```{r}
sub12<-pp_meta2 %>% filter( (PCx =="PC-01") & (PCy =="PC-02") ) 
                     
sub12 %>% group_by(Subspecies) %>% summarize(Count=n())

rb12<-ggplot(sub12) +
  geom_point(aes(x = val_x, y = val_y, fill = Watershed, shape=MajorGroup), size = 2, alpha=0.75) +
  scale_fill_discrete(na.value = "white") + 
  theme_bw()+
  theme(panel.grid=element_blank())+
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = ""))+
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = ""))+
#  scale_fill_viridis_d(option = "magma") +
  scale_shape_manual("Interior Redband Lineage", values = c(24,25,23)) +
  theme(plot.title = element_text(face="bold", size=16, hjust=0.5)) +
  guides(fill = guide_legend(override.aes=list(shape=22))) +
  ggtitle("Interior Redband Lineages Forward Only PCA")


rb12

ggsave("outputs/population-genetics-single/259.PC12.redb.jpeg")
```

```{r}
sub13<-pp_meta2 %>% filter( (PCx =="PC-01") & (PCy =="PC-03") )

rb13<-ggplot(sub13, aes(x = val_x, y = val_y, fill = Watershed, shape=MajorGroup)) +
  geom_point(size = 2, alpha=0.75) +
  scale_fill_discrete(na.value = "white") + 
  theme_bw()+
  theme(panel.grid=element_blank())+
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = ""))+
  ylab(paste("PC3", " ", round((100*var[3]),2), "%", sep = ""))+
  scale_shape_manual("Major Group", values = c(24,25)) +
  guides(fill = guide_legend(override.aes=list(shape=22))) +
  theme(plot.title = element_text(face="bold", size=16))



```

